compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -710,6 +705,63 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         )
     }
 
+    fn construct_suggestion(&self, err: &mut DiagnosticBuilder<'_, ErrorGuaranteed>, local: Local) {
+        let local_decl = &self.body.local_decls[local];
+        debug!("local_decl: {:?}", local_decl);
+        let pat_span = match *local_decl.local_info() {
+            LocalInfo::User(BindingForm::Var(mir::VarBindingForm {
+                binding_mode: ty::BindingMode::BindByValue(Mutability::Not),
```

Comment:
Probably not in scope for this PR, but we really should put a `HirId` in here.

Reply:
> Probably not in scope for this PR, but we really should put a `HirId` in here.

Thanks a lot, since this will cause multiple changes when using `LocalDecl`, I thought I could submit another pr to support this.